src/orion/algo/pbt/pbt.py

```diff
@@ -377,7 +370,14 @@ def _sample(self, num: int) -> list[Trial]:
 
     def _get_depth_of(self, fidelity: Any) -> int:
         """Get the depth of a fidelity in the lineages"""
-        return self.fidelities.index(fidelity)
+        # BUG: Issue with rounding, asking for fidelity of `10` but list has `10.00000004`.
```

Comment:
Is this bug fixed by this if-else?

Reply:
Yes, the if/else fixes it.
Can I leave this note ?

Reply:
(since I don't think this is particularly clean, it's more like a patch for a bug)

Reply:
Leaving the note is fine, but it should be more clear that this is a hacky bug fix, rather than a bug to be fixed.

Reply:
Still unaddressed. It's noting critical, but it's confusing.